File: LeetCode 30 Day Challenge/13-contiguous_array.py
class Solution(object):
    def findMaxLength(self, nums):
        dict = {0: -1}
        count = 0
        max_len = 0
        
        for i in range(len(nums)):
            if nums[i] == 0:
                count -= 1
            else:
                count += 1
            
            if count in dict:
                max_len = max(max_len, i - dict[count])
            else:
                dict[count] = i
        
        return max_len
        

File: LeetCode 30 Day Challenge/13-contiguousArray.js
const findMaxLength = (nums) => {
    let hash = {0: -1}
    let count = 0
    let max = 0
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 0) count--
        else count++
        
        if (hash[count] == null) hash[count] = i
        else max = Math.max(max, i - hash[count])
    }
    return max
}